src/FileWriter.php

```diff
@@ -25,13 +29,18 @@ final class FileWriter
     public static function writeFile(string $file, string $content, int $chmod = 0666) : void
     {
         $dir = dirname($file);
+
         // suppress notice thrown when falling back to system temp dir
-        $tmp = @tempnam($dir, 'wsw');
+        $tmp = self::suppress(static function () use ($dir) {
```

Comment:
The entire `set_error_handler()` stuff is a bit expensive - perhaps wrap everything within one big `self::suppress()`?